securedrop/journalist.py

```diff
@@ -242,9 +242,17 @@ def admin_reset_two_factor_hotp():
     otp_secret = request.form.get('otp_secret', None)
     if otp_secret:
         user = Journalist.query.get(uid)
-        user.set_hotp_secret(otp_secret)
-        db_session.commit()
-        return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        try:
+            user.set_hotp_secret(otp_secret)
+            db_session.commit()
+            return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        except TypeError as e:
```

Comment:
Since the `TypeError` we're trying to catch is going to be raised by `user.set_hotp_secret()`, and not by the following two lines it makes sense to put the two following lines in an `else` block.